### PR TITLE
Fix for Java 16+ CLI

### PR DESCRIPTION
**Pull request by shashachu**
## Description

Conditionally add `--add-opens` flag based on java version

Fixes https://github.com/pinterest/ktlint/issues/1195

## Checklist

- [ x ] `CHANGELOG.md` is updated

### CODE REVIEW

File: ktlint/build.gradle [hunk 1]
```diff
@@ -58,7 +58,7 @@ def shadowJarExecutableTask = tasks.register("shadowJarExecutable", DefaultTask.
   doLast {
     File execFile = outputs.files.getFiles().first()
     execFile.withOutputStream {
-      it.write "#!/bin/sh\n\nexec java -Xmx512m -jar \"\$0\" \"\$@\"\n\n".bytes
+      it.write "#!/bin/sh\n\nJV=\$(java -version 2>&1 | head -1 | cut -d'\"' -f2 | sed '/^1\\./s///' | cut -d'.' -f1)\n\nX=\$( [ \"\$JV\" -ge \"16\" ] && echo \"--add-opens java.base/java.lang=ALL-UNNAMED\" || echo \"\")\n\nexec java \$X -Xmx512m -jar \"\$0\" \"\$@\"\n\n".bytes
```
Comment (paul-dingemans) [inline ktlint/build.gradle:61]: Maybe it is an idea to add some comments (and the bash script without espace characters) here.

```
#!/bin/sh
JV=$(java -version 2>&1 | head -1 | cut -d'"' -f2 | sed '/^1\\./s///' | cut -d'.' -f1)
X=$( [ "$JV" -ge "16" ] && echo "--add-opens java.base/java.lang=ALL-UNNAMED" || echo "")
exec java $X -Xmx512m -jar "$0" "$@"
```

**Review by paul-dingemans — COMMENTED** [review]
I think this will work. Unfortunately, I am not able to reproduce any of the Java16 issues anymore on my local machine with the current version of ktlint with java 16 nor 17. So I can not test whether your fix solve those problems.

File: ktlint/build.gradle (hunk 1 above)
Comment (shashachu) [reply ktlint/build.gradle:61]: That's a good idea. I'm able to repro and confirm this fixes it, but @romtsn pointed out this likely won't work on Windows. I don't have a Windows machine to test on.

Comment (paul-dingemans) [reply ktlint/build.gradle:61]: Actually I had same doubt about windows. But the old code already uses a bash shell. The other utilities like cut and head are rather basic and most likely will work. But we won't know for sure untill tried. I have an old windows machine that I used for development years ago. There I used a tool called git-bash which supported all basic Unix tools.

Comment (shashachu) [reply ktlint/build.gradle:68]: I've updated it with comments. I think what I'll do is merge it and then we can ask folks on the bug thread to test on Windows with the SNAPSHOT.

**Pull request merged**